app/uk/gov/hmrc/residencenilratebandcalculator/controllers/SimpleControllerBase.scala

```diff
@@ -46,30 +47,44 @@ trait SimpleControllerBase[A] extends ControllerBase[A] {
 
   def form: () => Form[A]
 
-  def view(form: Option[Form[A]], backUrl: String)(implicit request: Request[_]): HtmlFormat.Appendable
+  def view(form: Option[Form[A]], backUrl: String, answerRows: Seq[AnswerRow])(implicit request: Request[_]): HtmlFormat.Appendable
 
   val navigator: Navigator
 
+  def answerRows(cacheMap: CacheMap, request: Request[_]): Seq[AnswerRow] = AnswerRows.constructAnswerRows(
+    AnswerRows.truncateAndLocateInCacheMap(controllerId, cacheMap),
+    AnswerRows.answerRowFns,
+    AnswerRows.rowOrder,
+    messagesApi.preferred(request)
+  )
+
   override def onPageLoad(implicit rds: Reads[A]): Action[AnyContent] = Action.async { implicit request =>
     sessionConnector.fetch().map {
       case None => Redirect(uk.gov.hmrc.residencenilratebandcalculator.controllers.routes.SessionExpiredController.onPageLoad())
       case Some(cacheMap) => {
-        Ok(view(cacheMap.getEntry(controllerId).map(value => form().fill(value)), navigator.lastPage(controllerId)(new UserAnswers(cacheMap)).url))
+        val previousAnswers = answerRows(cacheMap, request)
+        Ok(view(cacheMap.getEntry(controllerId).map(value => form().fill(value)),
+          navigator.lastPage(controllerId)(new UserAnswers(cacheMap)).url, previousAnswers))
       }
     }
   }
 
   def onSubmit(implicit wts: Writes[A]) = Action.async { implicit request =>
+
     sessionConnector.fetch().flatMap {
       case None => Future.successful(Redirect(uk.gov.hmrc.residencenilratebandcalculator.controllers.routes.SessionExpiredController.onPageLoad()))
       case Some(cacheMap) => {
+        val previousAnswers = answerRows(cacheMap, request)
```

Comment:
Can we either put this inline or pull out
new UserAnswers(cacheMap).url
for consistency?
I prefer in-lining it since we're only using it once so there's no computation saving to be had